tests/httpwrappers/tests.py

```diff
@@ -699,6 +703,42 @@ def test_repeated_nonstandard_keys(self):
         """
         self.assertIn('good_cookie', parse_cookie('a:=b; a:=c; good_cookie=yes').keys())
 
+    def test_python_cookies(self):
+        # from Python's Lib/test/test_http_cookies.py
+        self.assertEqual(parse_cookie('chips=ahoy; vienna=finger'), {'chips': 'ahoy', 'vienna': 'finger'})
+
+        # here we differ from how python parses cookies
+        self.assertEqual(
+            parse_cookie('keebler="E=mc2; L=\\"Loves\\"; fudge=\\012;"'),
+            {'keebler': '"E=mc2', 'L': '\\"Loves\\"', 'fudge': '\\012', '': '"'})
+
+        # Check illegal cookies that have an '=' char in an unquoted value
+        self.assertEqual(parse_cookie('keebler=E=mc2'), {'keebler': 'E=mc2'})
+
+        # Cookies with ':' character in their name.
+        self.assertEqual(parse_cookie('key:term=value:term'), {'key:term': 'value:term'})
+
+        # Cookies with '[' and ']'
+        self.assertEqual(parse_cookie('a=b; c=[; d=r; f=h'), {'a': 'b', 'c': '[', 'd': 'r', 'f': 'h'})
+
+    def test_cookie_garbage(self):
+        # total garbage that goes against rfc6265 but browsers will send if set via document.cookie
+        # unnamed cookies https://bugzilla.mozilla.org/show_bug.cgi?id=169091
+        self.assertIn('django_language', parse_cookie('abc=def; unnamed; django_language=en').keys())
+        # yes, even a double quote
+        self.assertEqual(parse_cookie('a=b; "; c=d'), {'a': 'b', '': '"', 'c': 'd'})
+        # spaces in names and values, equals in values
+        self.assertEqual(parse_cookie('a b c=d e=f; gh=i'), {'a b c': 'd e=f', 'gh': 'i'})
+        # repeated cookies (the cookie with the longer path goes first).
+        # We've historically only kept the last of multiples.
+        self.assertEqual(parse_cookie('a=b; h=i; a=c'), {'a': 'c', 'h': 'i'})
+        # stuff the spec explicitly forbids
+        self.assertEqual(parse_cookie('a   b,c=d  "  =e,f g'), {'a   b,c': 'd  "  =e,f g'})
+        # unicode, again the spec only allows ascii
+        self.assertEqual(parse_cookie('saint=André Bessette'), {'saint': smart_str('André Bessette')})
+        # browsers won't send extra whitespace or semicolons, but match behavior of document.cookie
```

Comment:
I think you mean "parse_cookie() should match the behavior of document.cookie" but the current phrasing it a bit ambiguous if "match the behavior" refers back to "browsers". Also wrap, the comments at 79 characters please.